src/Hl7.Fhir.Base/Serialization/BaseFhirJsonPocoSerializer.cs

```diff
@@ -151,7 +150,12 @@ private static string addSuffixToElementName(string elementName, object elementV
                 _ => null
             };
 
-            return typeName is null ? elementName : elementName + char.ToUpperInvariant(typeName[0]) + typeName.Substring(1);
+            // determine type by name, when not found use the first type
+            Type? type = types.FirstOrDefault(t => t.Name.Equals(typeName, StringComparison.OrdinalIgnoreCase)) ?? types.FirstOrDefault();
```

Comment:
The BaseFhirXmlPocoSerializer has exactly the same bug still in it.

Reply:
The good news is: this `requiredType` for which you are figuring out the type is *only* relevant and used in the case of Integer64, in which case the .NET and the FHIR typename are the same (modulo the casing).

Reply:
So, although the algorithm is buggy, the end effect is that is will work. We might make that explicit by checking explicitly for Integer64 - and if it is integer64 THEN passing typeof(integer64) as the requiredType parameter.

Reply:
Or just instead of:

```
  var propertyName = propertyMapping?.Choice == ChoiceType.DatatypeChoice ?
                            addSuffixToElementName(member.Key, member.Value) : member.Key;
  var requiredType = propertyMapping?.FhirType.FirstOrDefault();
```

this should work too:

```
     var propertyName = propertyMapping?.Choice == ChoiceType.DatatypeChoice ?
                            addSuffixToElementName(member.Key, member.Value) : member.Key;
     var requiredType = member.Value is Integer64 ? typeof(Integer64) : null;
```

Reply:
> The good news is: this `requiredType` for which you are figuring out the type is _only_ relevant and used in the case of Integer64, in which case the .NET and the FHIR typename are the same (modulo the casing).

Why is this only relevant for `Integer64`? Because we handle this type differently in Json (string instead of integer)? 
If the answer is yes, then I am in favor for just checking it for integer64 instead of taking the first member of the Type list.